### ISSUE TITLE
Seeding question

### ISSUE THREAD

**Issue by seabbs** [adrian-lison/EpiSewer#5]
> seeding: The renewal model used by EpiSewer requires a seeding phase during which the reproduction number cannot be modeled. We thus let the initial infections be estimated using a simple random walk model.

Is this a geometric random walk or is it a random walk on the growth rate? Have you played around with some different choices here?

**Comment by adrian-lison** [comment]
Yes, `seeding_estimate_rw()` models a geometric random walk on the expected number of infections. There is also `seeding_estimate_constant()`, which keeps the expected number of infections constant over the seeding phase. In both cases, realized infections are also subject to infection noise (if specified, which is the default). Interestingly, I haven't seen any relevant differences in Rt estimates beyond the seeding phase between the two. Also no big difference w.r.t sampling speed.

Based on that, I don't expect the growth rate version to make a big difference. But it might still be nice to support that option (maybe just estimating a fixed growth rate will suffice?).

**Comment by seabbs** [comment]
> Interestingly, I haven't seen any relevant differences in Rt estimates beyond the seeding phase between the two. Also no big difference w.r.t sampling speed.

Yeah I wouldn't expect it to make a large difference. A interesting choice @SamuelBrand1 and co made in EpiAware is to calculate the growth rate that the first Rt estimates represents and then use that for seeding which I think is quite nice.

 think your right though that your current default to something flexible with a few more parameters is likely safe and efficient enough in nearly all instances

**Comment by adrian-lison** [comment]
> A interesting choice @SamuelBrand1 and co made in EpiAware is to calculate the growth rate that the first Rt estimates represents and then use that for seeding which I think is quite nice.

Oh, this is indeed a very nice option! I will have to check this out, because afaik obtaining the growth rate from Rt is not trivial if the generation time distribution doesn't have a certain parametric form / is just a categorical distribution over days?

**Comment by SamuelBrand1** [comment]
> > A interesting choice @SamuelBrand1 and co made in EpiAware is to calculate the growth rate that the first Rt estimates represents and then use that for seeding which I think is quite nice.
> 
> Oh, this is indeed a very nice option! I will have to check this out, because afaik obtaining the growth rate from Rt is not trivial if the generation time distribution doesn't have a certain parametric form / is just a categorical distribution over days?

Yeah, we set it up as a root-finding problem to the discrete time renewal process and do it with a few quick Newton-Raphson steps which plays ok with autodiff.

**Comment by SamuelBrand1** [comment]
The main conceptual downside I think is that it assumes that your first estimates of $R_t$ are a continuation of some trend... whereas one can imagine situations where you might be interested in (say) flat incidence trend and then <insert reason> causes Rt to vary. We might well want to support that explicitly in the future.

**Comment by seabbs** [comment]
Yes, agree. I do like it as a baseline model to then build other features around though (i.e use this as the intercept and put a random effect in as well to allow initial variation).

**Comment by seabbs** [comment]
All this being said I think someone (?? any takers) really needs to do some eval work to see if any of these choices really influence things (and when they do) could make a nice little paper (though very specific)

**Comment by adrian-lison** [comment]
Thanks @SamuelBrand1, that sounds like a cool solution!

**Comment by adrian-lison** [comment]
@seabbs @SamuelBrand1 Thanks again for your input.

Follow up: I implemented the backcalculation of the growth rate based on the initial Rt as in EpiAware, i.e. also via root finding as you suggested @SamuelBrand1. It works pretty well.

I have now also added a backward-in-time random walk on the growth rate, which means that the growth rate will match the initial Rt at the end of the seeding phase, but it can vary before that and will generally be more uncertain towards the start of the seeding phase, which seems sensible.

I have done some very shallow testing, it seems like this growth rate option leads to slighlty slower sampling in stan than the simple random walk on log infections option. There is also no difference in Rt estimates outside of the seeding phase. Nevertheless, this new seeding model is quite useful as EpiSewer has an option to extend the seeding phase when wastewater detections are still sparse. For such a potentially longer seeding phase, being able to model exponential growth is important. Using the above approach we can ensure that the modeled transmission dynamics during that phase will be consistent with the intercept of the Rt model.

**Comment by seabbs** [comment]
> backward-in-time random walk on the growth rate

This is a really cool idea!
